rdup-tr.c

```diff
@@ -20,8 +20,16 @@ char *template;
 gboolean opt_tty = FALSE;	/* force write to tty */
 #ifdef HAVE_LIBNETTLE
 gchar *opt_crypt_key = NULL;	/* encryption key */
+guint opt_crypt_key_length = 0;
 gchar *opt_decrypt_key = NULL;	/* decryption key */
-struct aes_ctx *aes_ctx = NULL;
+guint opt_decrypt_key_length = 0;
+/*
+ * TODO: I don't like having 3 AES contexts, when only 1 is actually used (16
```

Comment:
Is it a stretch to just allow only one key-size?

Reply:
Actually that would remove pretty much all of the key size checking and make this change much simpler and shorter. 

We could just pick, say AES-256, and pad the keys to 32 bytes. There's two ways we could do that:
1. Pad with some pattern
2. Use something like PBKDF2 (preferable since it's more secure)

Reply:
agreed, let's do that. So to recap:
1. 1 keysze
2. pad with pbkdf2